modules/resource_group_storage/main.tf

```
@@ -0,0 +1,30 @@
+# Створення Resource Group
+resource "azurerm_resource_group" "main" {
+  name     = var.resource_group_name
+  location = var.location
+  
+  tags = var.tags
+}
+
+# Створення Storage Account
+resource "azurerm_storage_account" "main" {
+  name                     = var.storage_account_name
```

You reference storage_account_name here. Azure storage account names must be lowercase, alphanumeric and 3-24 characters. Consider adding a validation rule in variables.tf to enforce these constraints so invalid names fail fast and return a helpful error message.